conda_build/api.py

```diff
@@ -154,29 +155,47 @@ def get_output_file_paths(
             finalize=True,
             **kwargs,
         )
-    else:
-        assert hasattr(
-            recipe_path_or_metadata, "config"
-        ), f"Expecting metadata object - got {recipe_path_or_metadata}"
+
+    elif hasattr(recipe_path_or_metadata, "config"):
         metadata = [(recipe_path_or_metadata, None, None)]
-    #    Next, loop over outputs that each metadata defines
+
+    elif isinstance(recipe_path_or_metadata, Iterable) and all(
+        isinstance(recipe, tuple)
+        and len(recipe) == 3
```

Comment:
Should we consider removing the magic number for a `Final[int]`?

Reply:
Refactored to use a `NamedTuple` (`MetaDataTuple`) instead.